### DISCUSSION TITLE
debug plugin

### DISCUSSION THREAD

**Discussion by hammyau** [eclipse-tycho/tycho#800] [Q&A] [answered]
Hi, I have an eclipse tycho project that builds and the resulting exe runs just fine.

But what is the magic to be able to debug it with an Eclipse IDE?

Somehow I did manage it a few days ago but I am unclear what I did. I seem to recall the Run Configuration needs some magic.
I imported the whole project as a maven it builds fine via the Eclipse Maven too.

As an aside there is the mvn eclipse:eclipse plugin. But this is now deprecated?

I am missing something in terms of getting to debug my application.

Any pointers?

Thanks in advance.

**Comment by laeubi** [comment]
If you have a eclipse-product file, you can simply choose "Debug As Eclipse Application" there is a button in the product editor for it, if you wan't to debug a running "exe" then remote-debugging would be the way to go. Both techniques are independent from tycho though...

**Reply by hammyau** [reply]
Hi Christoph, thanks for the fast reply. I have tried that but it says there are missing bundles.
I just noticed that the target platform is using latest in the location. I am changing it to the IDE version I am using 2021-09.
Maybe that will help.

**Reply by laeubi** [reply]
You should set the same target platform that you are using for your tycho build.

**Reply by hammyau** [reply]
The target platform now matches. Just rebuilt. But it is still reporting missing items via the product file start.

For example....
![image](https://user-images.githubusercontent.com/1050446/159883922-23fe00f5-456c-42c3-8411-3c71baf9c6c8.png)

**Comment by mickaelistria** [comment, marked as answer]
If you want to debug as you're developing your application, then use PDE debug capabilities as suggested by Christoph. But this does require you to properly configure your workspace to develop your application.
If you want to debug the resulting application, just tweak the <app>.ini file of your application to add some debugging VM args (standard Java flags), and then connect to it with https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-remotejava_launch_config.htm

**Reply by hammyau** [reply]
Thanks guys, the remote debugging option is working for me,
